src/pipeline.ts

```diff
@@ -101,6 +100,14 @@ export interface GitHubWorkflowProps extends PipelineBaseProps {
    * you will be logged in to docker when you upload Docker Assets.
    */
   readonly dockerCredentials?: DockerCredential[];
+
+  /**
+   * The type of machine to run the job on. The machine can be either a
+   * GitHub-hosted runner or a self-hosted runner.
+   *
+   * @default "ubuntu-latest"
+   */
+  readonly runsOn?: string;
```

Comment:
lets make this type an enum-like class `Runner` that looks something like this. I'm modeling this off of the aws-cdk lambda [`Runtime`](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-lambda/lib/runtime.ts) that we have. I haven't fully thought this through but I'd like it to be modeled something like this:

```ts
export class RuntimeType {
  public static readonly GITHUB_HOSTED = 'github-hosted';
  public static readonly SELF_HOSTED = 'self-hosted';
}

export class Runtime {
  public static readonly UBUNTU_LATEST =  new Runner(['ubuntu-latest'], RuntimeType.GITHUB_HOSTED);

  public static readonly WINDOWS_LATEST = new Runner(['windows-latest'], RuntimeType.GITHUB_HOSTED);

  public static readonly MACOS_LATEST = new Runner(['macos-latest'], RuntimeType.GITHUB_HOSTED);

  public static selfHosted(labels: string[]): Runner {
    return new Runtime(['self-hosted', ...labels]);
  } 

  public constructor(public readonly runsOn: string) {}
}
```

One thing we'll have to figure out is how to get `runsOn: ubuntu-latest` most of the time, but if you want a self-hosted runtime you can have a [list](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-lambda/lib/runtime.ts) of labels. Again, I haven't fully thought this through, so feel free to make some changes to the model I'm suggesting.

Reply:
Thanks for the input @kaizen3031593. I've updated the PR.